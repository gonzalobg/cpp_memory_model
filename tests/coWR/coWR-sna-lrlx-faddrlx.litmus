C coWR-sna-lrlx-faddrlx
{ [x] = 0; }

P0 (int* x) {
  *x = 1;
  int a = atomic_load_explicit(x, memory_order_relaxed);
}

P1 (int* x) {
  atomic_fetch_add_explicit(x, 2, memory_order_relaxed);
}

~exists (0:a=2 /\ [x]=1)