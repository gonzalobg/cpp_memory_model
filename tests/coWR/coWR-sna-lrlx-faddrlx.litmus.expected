Test coWR-sna-lrlx-faddrlx Forbidden
States 3
0:a=1; [x]=1;
0:a=1; [x]=3;
0:a=3; [x]=3;
Undef
Witnesses
Positive: 3 Negative: 0
Flag *undef*
Condition ~exists (0:a=2 /\ [x]=1)
Observation coWR-sna-lrlx-faddrlx Never 0 3
Hash=9ef3b5d7099535f3b52149e94006a5ab

